packages/rum-core/src/browser/htmlDomUtils.ts

```diff
@@ -0,0 +1,31 @@
+export function isTextNode(node: Node): node is Text {
+  return node.nodeType === Node.TEXT_NODE
+}
+
+export function isCommentNode(node: Node): node is Comment {
+  return node.nodeType === Node.COMMENT_NODE
+}
+
+export function isElementNode(node: Node): node is Element {
+  return node.nodeType === Node.ELEMENT_NODE
+}
+
+export function isNodeShadowHost(node: Node): node is Element & { shadowRoot: ShadowRoot } {
+  return isElementNode(node) && node.shadowRoot !== null
+}
+
+export function isNodeShadowRoot(node: Node): node is ShadowRoot {
+  const shadowRoot = node as ShadowRoot
+  return !!shadowRoot.host && isElementNode(shadowRoot.host)
+}
+
+export function getChildNodes(node: Node) {
+  return isElementNode(node) && node.shadowRoot ? node.shadowRoot.childNodes : node.childNodes
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
  return isNodeShadowHost(node) ? node.shadowRoot.childNodes : node.childNodes
```